lib/internal/modules/esm/fetch_module.js

```diff
@@ -138,15 +138,16 @@ function isRedirect(statusCode) {
  * @param {URL} parsed
  * @returns {Promise<CacheEntry> | CacheEntry}
  */
-function fetchWithRedirects(parsed) {
+function fetchWithRedirects(parsed, context) {
   const existing = cacheForGET.get(parsed.href);
   if (existing) {
     return existing;
   }
   const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
   const result = (async () => {
+    const type = context.importAttributes?.type;
     const req = handler(parsed, {
-      headers: { Accept: '*/*' },
+      headers: { Accept: type === 'json' ? 'application/json,*/*;q=0.5' : '*/*;' },
```

Comment:
There are more than only `json` planned for import attributes, so I think it would be better to make a dict for this, like:

```js
const accept = acceptMimes[context.importAttributes?.type] ?? acceptMimes.default;
const req = handler(parsed, {
      headers: { Accept: accept },
```

```js
const acceptMimes = {
  __proto_: null,
  default: '*/*',
  json: 'application/json,*/*;q=0.5',
};
```